posthog/temporal/batch_exports/transformer.py

```diff
@@ -225,4 +301,37 @@ def write_batch(self, record_batch: pa.RecordBatch) -> typing.Generator[bytes, N
         yield data
 
 
+def transform_record_batch_from_shared_memory(shm_name: str, transformer: StreamTransformer):
+    """Top level function to run transformers in multiprocessing.
+
+    Record batches are accessed from shared memory.
+    """
+    shm = None
+    record_batch = None
+    buffer_reader = None
+    reader = None
+
+    try:
+        shm = sm.SharedMemory(name=shm_name)
+        buffer_reader = pa.BufferReader(shm.buf)
+        with pa.RecordBatchStreamReader(buffer_reader) as reader:
+            record_batch = reader.read_next_batch()
+
+        record_batch = cast_record_batch_json_columns(record_batch)
+        chunks = list(transformer.transform_batch(record_batch))
+
+        return chunks
+    finally:
+        if buffer_reader:
+            del buffer_reader
+        if record_batch:
+            del record_batch
+        if reader:
+            del reader
```

Comment:
are these needed, since once the process terminates, I presume these would all be deleted anyway?

Reply:
And we need to close it + unlink it otherwise we leak memory.

Reply:
Maybe there is a more elegant way to do this, like hiding it behind a context manager.